keys.go

```diff
@@ -461,6 +464,62 @@ func (c *Client) Rotate(ctx context.Context, id, payload string) error {
 	return nil
 }
 
+type KeyPayload struct {
+	payload             string
+	encryptedNonce      string
+	iv                  string
+	encryptionAlgorithm string
+}
+
+func NewKeyPayload(payload, encryptedNonce, iv string) KeyPayload {
+	kp := KeyPayload{
+		payload:        payload,
+		encryptedNonce: encryptedNonce,
+		iv:             iv,
+	}
+	return kp
+}
+
+// EncryptWithRSA256 sets the encryption algorithm for key create to RSAES_OAEP_SHA_256
+// This is the default algorithm for key creation under Key Protect service
+func (kp KeyPayload) WithRSA256() KeyPayload {
+	kp.encryptionAlgorithm = "RSAES_OAEP_SHA_256"
+	return kp
+}
+
+// EncryptWithRSA1 sets the encryption algorithm for key create to RSAES_OAEP_SHA_1
+// This algorithm is only supported by the Hyper Protect(HPCS) service
+func (kp KeyPayload) WithRSA1() KeyPayload {
+	kp.encryptionAlgorithm = "RSAES_OAEP_SHA_1"
+	return kp
+}
+
+// Rotate2 methods supports rotation of a root key with or without payload and also rotate a
+// securely imported root key.
+func (c *Client) Rotate2(ctx context.Context, id string, new_key *KeyPayload) error {
```

Comment:
Good to rename this to something other than Rotate2
may be `RotateKeys`?

Reply:
`Rotate2` follow golang convention to have an overloaded or superseded method, since there is no overloading built into the language...

We do this for unwrap already.

Reply:
@mrodden should we go for this method name `RotateV2`

Reply:
@pbavari renaming the method to `RotateV2` instead.

Reply:
if it is golang convention then we can go with it to make it consistent with other methods we have.

Reply:
We set it consistent with our methods. We have UnwrapV2 so to match that RotateV2